crates/ide-assists/src/handlers/extract_module.rs

```diff
@@ -439,7 +439,28 @@ impl Module {
                                         ctx,
                                     )
                                 {
-                                    import_paths_to_be_removed.push(import_path);
+                                    if import_paths_to_be_removed.len() > 0 {
+                                        // Text ranges recieved here for imports are extended to the
+                                        // next/previous comma which can cause intersections among them
+                                        // and later deletion of these can cause panics similar
+                                        // to reported in #11766. So to mitigate it, we
+                                        // check for intersection between all current members
+                                        // and if it exists we combine both text ranges into
+                                        // one
+                                        for i in 0..import_paths_to_be_removed.len() {
+                                            if let Some(_) =
+                                                import_paths_to_be_removed[i].intersect(import_path)
+                                            {
+                                                import_paths_to_be_removed[i] =
+                                                    import_paths_to_be_removed[i]
+                                                        .cover(import_path);
+                                            } else {
+                                                import_paths_to_be_removed.push(import_path);
```

Comment:
Should this only run if there is no intersection in the entire loop? This seems to dupe import_path with each iteration if there is no intersection.

Reply:
Hey @jhgg , thanks for pointing this out, not sure how I missed this. I have pushed the changes to it with the new commit and also factored it out in a function because I forgot to add it for `names`, if you have a look at surrounding code it has two cases of `names` and `namerefs`. I added it for `namerefs`, forgot to do the same for `names`, tho everything is fixed now.